src/Lib.hs

```diff
@@ -91,28 +187,31 @@ data Pet = Pet
   { petName :: Name
   , petAttack :: Attack
   , petHealth :: Health
+  , petHealthRemaining :: Health
   , petCost :: Cost
   }
--- TODO how to do this? is this the best way to do this? is this necessary?
--- data AnimalType = Turtle | Ant | Penguin | etc...
 
+mkPet :: Name -> Attack -> Health -> Cost -> Pet
+mkPet name attack health cost = Pet {petName=name, petAttack=attack, petHealth=health, petHealthRemaining=health, petCost=cost}
```

Comment:
would suggest spaces around =, and maybe consider breaking this up into multiple lines as it is a bit long.